src/View.php

```diff
@@ -265,6 +265,10 @@ protected function init(): void
         $this->_add_later = [];
         parent::init();
 
+        if (!$this->executorFactory) {
+            $this->executorFactory = $this->getApp()->defaultExecutorFactory;
```

Comment:
I would highly prefer an instance here, so we do not need static at all

And - probably drop `executorFactory` in view, we do not want that for almost all render tree item.

If user should have option to customize this factory on per View instance basis - why? - then add getter with default to app here which can be overriden.

Reply:
> If user should have option to customize this factory on per View instance basis - why? - then add getter with default to app here which can be overriden.

Why not? Let says you have one specific $card using a lot of model action, you can then easily set this card only with buttons using icon only, while you would still need full button label on other $card view.

Reply:
if needed in multiple extended View classes, add with trait, View is base building block and I do not belive `View` class itself benefits by having executor factory in any way

Reply:
then annotate the property as object only (no string support)

Reply:
> if needed in multiple extended View classes, add with trait, View is base building block and I do not belive `View` class itself benefits by having executor factory in any way

Trait will not do any good, since executorFactory is needed within View::on() method.

Reply:
Let's take two simple UI elements:

- bare `View`
- a `Button`

how does having a property with executor factory helps on these elements?

Reply:
$btn-on('click', $model-getUserAction('test));

Reply:
You can still pass an action to on method. But since the action needs an executor, then we need ExecutorFactory

Reply:
Well, ok, let's have that property for now but do not assing by default. Always fallback to app - add a protected getter method for it like `return prop ?? app->...`.

Reply:
already done! :)

Reply:
nice! :)